### ISSUE TITLE
Building nixpkgs.icoutils on darwin fails

### ISSUE THREAD

**Issue by HerrSpace** [NixOS/nixpkgs#151338]
### Describe the bug

Building nixpkgs.icoutils on arm64 darwin fails with the following error:
`/nix/store/ypqm9riz1gr599jdv9sh45g1wfknnqsj-libSystem-11.0.0/include/stdlib.h:181:5: error: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]`

Haven't reproed, but [looks also broken on AMD64](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.icoutils.x86_64-darwin).

### Steps To Reproduce
1. Try to build it with something like `nix-build ~/repos/upstream/nixpkgs -A icoutils`
2. Get the error above. [Full build log here](https://gist.github.com/HerrSpace/982beec8915e207b485c745539ad979c#file-gistfile1-txt-L1738)

### Expected behavior
The build should succeed.

### Additional context
- It seems to me like TARGET_OS_IPHONE is defined [here in Libsystem](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/darwin/apple-source-releases/Libsystem/default.nix#L71)
- It's unclear to me how to reference Libsystem properly from icoutils.

### Notify maintainers
- Maybe the Libsystem maintainers can enlighten me here :) @copumpkin @gridaphobe 
- Also @matthewbauer who [marked icoutils as supported by darwin here](https://github.com/NixOS/nixpkgs/commit/fe9b9e7f38999ca219ad9e6b4cf6447320ec9b70)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.2.0, macOS 12.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixpkgs-22.05pre339191.e1f9e754a40"`
 - channels(space): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: nixpkgs.ghidra-bin
# a list of nixos modules affected by the problem
module:
```

**Comment by veprbl** [comment]
Right, I think this would not be a problem on x86_64-darwin. I suppose the aarch64-darwin may use a different definition for Libsystem.

cc @thefloweringash

**Comment by wegank** [comment]
Fixed by #161620.
